modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1853,36 +1902,68 @@ private CompletableFuture<Object> applyUpdateCommand(UpdateCommand cmd) {
      * @param cmd UpdateAll command.
      * @return Raft future, see {@link #applyCmdWithExceptionHandling(Command)}.
      */
-    private CompletableFuture<Object> applyUpdateAllCommand(UpdateAllCommand cmd) {
+    private CompletableFuture<CompletableFuture<?>> applyUpdateAllCommand(UpdateAllCommand cmd, boolean skipDelayedAck) {
         if (!cmd.full()) {
-            storageUpdateHandler.handleUpdateAll(
-                    cmd.txId(),
-                    cmd.rowsToUpdate(),
-                    cmd.tablePartitionId().asTablePartitionId(),
-                    rowIds -> txsPendingRowIds.compute(cmd.txId(), (k, v) -> {
-                        if (v == null) {
-                            v = new TreeSet<>();
-                        }
+            if (skipDelayedAck) {
+                storageUpdateHandler.handleUpdateAll(
+                        cmd.txId(),
+                        cmd.rowsToUpdate(),
+                        cmd.tablePartitionId().asTablePartitionId(),
+                        rowIds -> txsPendingRowIds.compute(cmd.txId(), (k, v) -> {
+                            if (v == null) {
+                                v = new TreeSet<>();
+                            }
 
-                        v.addAll(rowIds);
+                            v.addAll(rowIds);
 
-                        return v;
-                    }),
-                    null);
-        }
+                            return v;
+                        }),
+                        null);
+
+                return applyCmdWithExceptionHandling(cmd).thenApply(res -> null);
+            } else {
+                CompletableFuture<Object> fut = applyCmdWithExceptionHandling(cmd).thenApply(res -> {
+                    // Currently result is always null on a successfull execution of a replication command.
+                    // This check guaranties the result will never be lost.
+                    assert res == null : "Replication result is lost";
```

Comment:
Well, from my point of view it will always be null, because basically it's void.

Reply:
But the method signature says otherwise. The result can appear later, so I've added some protection.